tests/test_builtin.js

```diff
@@ -997,15 +997,19 @@ function test_finalization_registry()
     {
         let expected = {};
         let actual;
-        let finrec = new FinalizationRegistry(v => { actual = v });
-        finrec.register({}, expected); // {} goes out of scope immediately
-        assert(actual, expected);
+        let finrec = new FinalizationRegistry(v => { 
+            actual = v;
+            assert(actual, expected);
```

Comment:
This weakens the test because the assert won't run when the bug it tests for is reintroduced.

Node's test suite has a `mustCall()` method but it's implemented in terms of `process.on("exit")` and qjs doesn't have that, of course. Maybe you can sort of fake it with a `setTimeout()` but that might be flaky.

(As an aside: it's no longer necessary to have the `let actual` at the top; rename the parameter to `actual` and done)

Reply:
We have queueMicrotask, maybe a check there is good.